### DISCUSSION TITLE
issue with dynamic import in edge function

### DISCUSSION THREAD

**Discussion by saschthetasch** [supabase/supabase#18662] [Questions]
# Issue
I have one edge function with different imports at the top. To trigger this function we post a request with an "id" and "slug" to import dynamic the config based on this values. 

Local works everything well but if i deploy it to production the i got following message
![Bildschirmfoto 2023-11-02 um 11 07 15](https://github.com/supabase/supabase/assets/8791514/885e408c-b69b-4881-bcde-0ce4fd285f74)

My code look like this: 
```
....
try {
    // Grab a connection from the pool
    const connection = await pool.connect()
    
    try {
      const results = await connection.queryObject(sql)
      const config = await import(`../_shared/configs/${slug}-${id}/config.ts`)
...
```

In my local environment it works very well but after i deploy it, i got everytime the error above.

# my thoughts
My thesis is, that supabase edge functions dont allow dynamic imports. But hopefully someone can help me

**Comment by anupriya17** [comment]
We're in the same boat but its not entirely supabase its Deno issue. 
https://docs.deno.com/runtime/manual/tools/compiler#dynamic-imports

**Comment by anupriya17** [comment]
After looking at the code, found a hack.
Import these files somewhere, probably all imports in a single file.
```
import * as slug1 from '../_shared/configs/slug-1/config.ts'
import * as slug2 from '../_shared/configs/slug-2/config.ts'
import * as slug3 from '../_shared/configs/slug-3/config.ts'
```

Then import this file in the file you use dynamic import. Since non-statically analyzable dynamic imports won't work as per docs
